lib/patterns.rb

```diff
@@ -7,6 +7,9 @@ module Patterns
   ROUTE_PATTERN         = /#{ALLOWED_CHARACTERS}/
   LAZY_ROUTE_PATTERN    = /#{ALLOWED_CHARACTERS}?/
   NAME_PATTERN          = /\A#{ALLOWED_CHARACTERS}\z/
+  LETTERS_REGEXP        = /[a-zA-Z]+/
+  SPECIAL_CHAR_PREFIX_REGEXP = /\A[#{Regexp.escape(SPECIAL_CHARACTERS)}]+/
```

Comment:
Could drop the `+` at the end and maintain behavior and linear time.

Reply:
it should still be linear in input size with the `+`, no?

Reply:
Just simpler, the `+` does nothing here. That's all.